### PR TITLE
Move secrets package to buildah

### PR DESCRIPTION
**Pull request by rhatdan**
Trying to remove circular dependencies between libpod and buildah.

First step to move pkg content from libpod to buildah.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by mheon** [comment]
LGTM, but it's failing gating

File: vendor.conf [hunk 1]
```diff
@@ -93,7 +93,7 @@ k8s.io/apimachinery kubernetes-1.10.13-beta.0 https://github.com/kubernetes/apim
 k8s.io/client-go kubernetes-1.10.13-beta.0 https://github.com/kubernetes/client-go
 github.com/mrunalp/fileutils 7d4729fb36185a7c1719923406c9d40e54fb93c7
 github.com/varlink/go 3ac79db6fd6aec70924193b090962f92985fe199
-github.com/containers/buildah v1.7.1
+github.com/containers/buildah 7d244ec5bb9fe5a8873720fbfa42863cfe60dfe0
```
Comment (vrothberg) [inline vendor.conf]: This commit doesn't exist in master.

**Comment by TomSweeneyRedHat** [comment]
Assuming happy tests and the right git version for Buildah, LGTM

**Comment by umohnani8** [comment]
LGTM

**Pull request merged**